### PR TITLE
Fix `NavigationSchoolAnalyzer` and create unit tests

### PR DESCRIPTION
**Pull request by mmusich**
#### PR description:

In commit https://github.com/cms-sw/cmssw/commit/47393c900567f9665a02c56de215b3dc259a361b I have accidentally introduced a wrong `esConsumes`statement that prevents to use `NavigationSchoolAnalyzer`.
I fix it here in commit 24787a2e846966fcc27b01d30508a0333ef20a55 and well as splitting the configuration files for Run-3 and Phase-2 and introducing dedicated unit tests to avoid future issues.

#### PR validation:

`cmssw` compiles, run successfully `scram b runtests`.

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

N/A

Cc:
@bdanzi

### CODE REVIEW

**Comment by mmusich** [comment]
type bug-fix

**Comment by mmusich** [comment]
@cmsbuild, please test

File: RecoTracker/TkNavigation/test/NavigationSchoolAnalyzer_Phase2_cfg.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import FWCore.ParameterSet.Config as cms
+
+# set the geometry and the GlobalTag
+
+process = cms.Process("NavigationSchoolAnalyze")
+
+#process.load("Configuration.StandardSequences.GeometryDB_cff")
```
Comment (mandrenguyen) [inline RecoTracker/TkNavigation/test/NavigationSchoolAnalyzer_Phase2_cfg.py:7]: Please remove commented code, unless needed

File: RecoTracker/TkNavigation/test/NavigationSchoolAnalyzer_Phase2_cfg.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import FWCore.ParameterSet.Config as cms
+
+# set the geometry and the GlobalTag
+
+process = cms.Process("NavigationSchoolAnalyze")
```
Comment (mandrenguyen) [inline RecoTracker/TkNavigation/test/NavigationSchoolAnalyzer_Phase2_cfg.py:5]: Missing an `r` at the end?

Comment (mmusich) [reply RecoTracker/TkNavigation/test/NavigationSchoolAnalyzer_Phase2_cfg.py:5]: really?

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by mandrenguyen** [comment]
+reconstruction

**Comment by perrotta** [comment]
+1

**Pull request merged**